lib/business_time/currency.rb

```diff
@@ -0,0 +1,24 @@
+module BusinessTime::Currency
+  # @return [Array] the list of currencies
+  # @param [String+]
+  def args(*currency)
+    return nil if currency.length.zero?
+    
+    curr = if currency.length == 1 # string
+              if currency.first.length.remainder(3).zero? # 'EURUSD'
+                currency.first.scan /.{3}/
+              elsif currency.first.length == 7           # 'EUR|USD'
+                currency.first.split '|'
+              else
+                [''] # will raise an error below, DRY
+              end
+           else
+             currency
+           end
+
+    raise ArgumentError.new("Wrong currency argument [#{currency}]") \
+      unless curr.all? { |it| String === it && it.length == 3 }
```

Comment:
Are you the first person besides me, who loves case-equal notation, or you just hesitated to change it to `it.is_a?(String)`?